src/java.base/share/classes/java/lang/StringConcatHelper.java

```diff
@@ -799,4 +725,110 @@ static String concat(String prefix, Object value, String suffix) {
         if (suffix == null) suffix = "null";
         return concat0(prefix, stringOf(value), suffix);
     }
+
+    @ForceInline
+    static String concat(String prefix, float value, String suffix) {
+        if (prefix == null) prefix = "null";
```

Comment:
Since we'll never bind in `null` values all these `prefix == null` are likely redundant unless we expose them to users. Which we probably shouldn't. It's a good thing this PR actually removes some shared secrets rather than adding new ones.

Reply:
String concatenation is required in many places in java.lang. These static concat methods will be used instead of "+", so null value processing is added. This is also the motivation for using static concat methods instead of Concat1.

Reply:
I don't think replacing a lot of concatenations in java.base with `SCH.concat` is very appealing and needs to be motivated by a substantial performance advantage. And for the places where it's motivated we can make sure to sanitize and handle `null` arguments.

Reply:
Yes, we shouldn't pay for things we don't need. I added null value handling to the parameters in SCF and removed it in SCH.